File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-8.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int res = 0;
        for(String token:tokens){
            if(!isOperator(token)){
                stack.push(Integer.parseInt(token));
            }
            else{
                if(token.equals("+")){
                    int num2 = stack.pop();
                    int num1 = stack.pop();
                    res = num1 + num2;
                }
                else if(token.equals("-")){
                    int num2 = stack.pop();
                    int num1 = stack.pop();
                    res = num1 - num2;
                }
                if(token.equals("*")){
                    int num2 = stack.pop();
                    int num1 = stack.pop();
                    res = num1 * num2;
                }
                if(token.equals("/")){
                    int num2 = stack.pop();
                    int num1 = stack.pop();
                    res = num1 / num2;
                }
                stack.push(res);
            }
        }
        return stack.pop();
        
    }
    private boolean isOperator(String token){
        return token.equals("+") ||
        token.equals("-") ||
        token.equals("*") ||
        token.equals("/");
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-8.py
class Solution:
    def isValid(self, s: str) -> bool:
        hm = {'(' : ')','{' : '}','[' : ']'}
        openB = []

        for i in range(len(s)):
            if s[i] in hm:
                openB.append(s[i])
            else:
                if not openB or s[i]!=hm[openB[-1]]:
                    return False
                openB.pop()
        
        return not openB